src/impl/apple/keychain_helper.cpp

```diff
@@ -109,21 +109,36 @@ void set_key(const std::vector<char>& key, CFStringRef account, CFStringRef serv
 
 }   // anonymous namespace
 
-std::vector<char> metadata_realm_encryption_key(bool check_legacy_service)
+std::vector<char> metadata_realm_encryption_key(bool check_legacy_service, bool recheck_legacy_service)
```

Comment:
Why did you leave this argument in place? The only caller always passes a constant value for it.

Reply:
I was thinking we might want to make it conditional in the future, or even remove it altogether. But maybe it makes more sense to just not have it at all.

Reply:
When we determine what to do we'll have to make a code change, so we can make it conditional at that point. As it stands, having it look like it's conditional when it's not just makes the code harder to follow (IMO).

Reply:
Yeah, I agree with you. I'll change it.